docs/build-docs.rst

```diff
@@ -0,0 +1,8 @@
+Building documentation locally
+==============================
+
+To build documentation locally, use:
+
+.. code-block:: bash
+
+    make docs-clean docs-html
```

Comment:
docs-clean is not required for building the docs for the first time. installing requirements, however, is required.
My suggestion:
- [ ] Create requirements-optional.txt
- [ ] Move flake8 & sphinx there
- [ ] Update this guide with 3 steps (install requirements, requirements-optional and `make docs-html`)
- [ ] Update `.travis.yml` accordingly